aten/src/ATen/native/TensorShape.cpp

```diff
@@ -631,19 +631,19 @@ static void check_t(const Tensor& self, const char *fn) {
              fn, " expects a tensor with 1 or 2 sparse and 0 dense dimensions, but got ",
              sparse_dim, " sparse and ", dense_dim, " dense dimensions");
   } else {
-    AT_CHECK((self.dim() == 1 || self.dim() == 2),
-             fn, " expects a 1D or 2D tensor, but self is ", self.dim(), "D");
+    AT_CHECK((self.dim() >= 0 && self.dim() <= 2),
```

Comment:
`self.dim() >= 0` is a tautology I think.

Reply:
Thanks @vishwakftw 
I wanted to ensure edge case of dim() being negative but it's not possible, hence removing.